### PR TITLE
Add support for HP PDU Management Module

### PR DESCRIPTION
**Pull request by Jellyfrog**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: includes/polling/os/hpe-pdumm.inc.php [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<?php
+/**
+ * hpe-ipdu.inc.php
```
Comment (PipoCanaja) [inline includes/polling/os/hpe-pdumm.inc.php]: we can feel the source of this file :)

**Comment by Jellyfrog** [comment]
How do I limit how many sensors it returns?
This unit reports 6 breakers per PDU:
```
.iso.org.dod.internet.private.enterprises.compaq.cpqPower.pdu.pduOutput.pduOutputBreakerTable.pduOutputBreakerEntry.breakerCurrent.1.1 = INTEGER: 3  
.iso.org.dod.internet.private.enterprises.compaq.cpqPower.pdu.pduOutput.pduOutputBreakerTable.pduOutputBreakerEntry.breakerCurrent.1.2 = INTEGER: 3  
.iso.org.dod.internet.private.enterprises.compaq.cpqPower.pdu.pduOutput.pduOutputBreakerTable.pduOutputBreakerEntry.breakerCurrent.1.3 = INTEGER: 1  
.iso.org.dod.internet.private.enterprises.compaq.cpqPower.pdu.pduOutput.pduOutputBreakerTable.pduOutputBreakerEntry.breakerCurrent.1.4 = INTEGER: 0  
.iso.org.dod.internet.private.enterprises.compaq.cpqPower.pdu.pduOutput.pduOutputBreakerTable.pduOutputBreakerEntry.breakerCurrent.1.5 = INTEGER: 0  
.iso.org.dod.internet.private.enterprises.compaq.cpqPower.pdu.pduOutput.pduOutputBreakerTable.pduOutputBreakerEntry.breakerCurrent.1.6 = INTEGER: 0  
.iso.org.dod.internet.private.enterprises.compaq.cpqPower.pdu.pduOutput.pduOutputBreakerTable.pduOutputBreakerEntry.breakerCurrent.2.1 = INTEGER: 2  
.iso.org.dod.internet.private.enterprises.compaq.cpqPower.pdu.pduOutput.pduOutputBreakerTable.pduOutputBreakerEntry.breakerCurrent.2.2 = INTEGER: 2  
.iso.org.dod.internet.private.enterprises.compaq.cpqPower.pdu.pduOutput.pduOutputBreakerTable.pduOutputBreakerEntry.breakerCurrent.2.3 = INTEGER: 1  
.iso.org.dod.internet.private.enterprises.compaq.cpqPower.pdu.pduOutput.pduOutputBreakerTable.pduOutputBreakerEntry.breakerCurrent.2.4 = INTEGER: 0  
.iso.org.dod.internet.private.enterprises.compaq.cpqPower.pdu.pduOutput.pduOutputBreakerTable.pduOutputBreakerEntry.breakerCurrent.2.5 = INTEGER: 0  
.iso.org.dod.internet.private.enterprises.compaq.cpqPower.pdu.pduOutput.pduOutputBreakerTable.pduOutputBreakerEntry.breakerCurrent.2.6 = INTEGER: 0  
```
But there is only 3 available, this is controlled by:
```
.iso.org.dod.internet.private.enterprises.compaq.cpqPower.pdu.pduOutput.pduOutputTable.pduOutputEntry.pduOutputNumBreakers.1 = INTEGER: 3            
.iso.org.dod.internet.private.enterprises.compaq.cpqPower.pdu.pduOutput.pduOutputTable.pduOutputEntry.pduOutputNumBreakers.2 = INTEGER: 3   
``` 

@PipoCanaja suggested:
```yaml
skip_values:
                        - 
                            oid: oidUpperLimit
                            op: '<='
                            value: '{{ $index }}'
```
Not really sure how to translate it to this, I probably need to add it to `pre_cached` ?

**Comment by PipoCanaja** [comment]
If the oid you need to compare to is not in the current table, then yes, you need to pre-cache it.

**Comment by Jellyfrog** [comment]
Seems It's not possible yet with yaml discovery.
I added:
```
    pre-cache:
        data:
            - oid:
                - pduOutputNumBreakers
```
And
```
                    skip_values:
                        -
                            oid: pduOutputNumBreakers
                            op: '<='
                            value: '{{ $subindex1 }}'
```

I logged the output here: https://github.com/librenms/librenms/blob/0b4c7b584a70ff4bf4bcd5ba2a660258d1218a96/LibreNMS/Device/YamlDiscovery.php#L255
And it returns:
```
NULL
string(16) "{{ $subindex1 }}"
```


The pre_cache array contains:
```
array(4) {
  ["pduOutputNumBreakers"]=>
  array(2) {
    [1]=>
    array(1) {
      ["pduOutputNumBreakers"]=>
      string(1) "3"
    }
    [2]=>
    array(1) {
      ["pduOutputNumBreakers"]=>
      string(1) "3"
    }
  }
  ["pduOutputBreakerEntry"]=>
  array(12) {
    ["1.1"]=>
    array(5) {
      ["breakerIndex"]=>
      string(1) "1"
      ["breakerVoltage"]=>
      string(3) "231"
      ["breakerCurrent"]=>
      string(1) "3"
      ["breakerPercentLoad"]=>
      string(2) "20"
      ["breakerStatus"]=>
      string(1) "0"
    }
    ["1.2"]=>
    array(5) {
      ["breakerIndex"]=>
      string(1) "2"
      ["breakerVoltage"]=>
      string(3) "232"
      ["breakerCurrent"]=>
      string(1) "3"
      ["breakerPercentLoad"]=>
      string(2) "21"
      ["breakerStatus"]=>
      string(1) "0"
    }
    ["1.3"]=>
    array(5) {
      ["breakerIndex"]=>
      string(1) "3"
      ["breakerVoltage"]=>
      string(3) "232"
      ["breakerCurrent"]=>
      string(1) "2"
...
```

**Comment by Jellyfrog** [comment]
@murrant could you confirm the above?

**Comment by PipoCanaja** [comment]
Let's start with this.

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/v1-58-release-changelog-november-2019/10224/1

**Pull request merged**